packages/filebrowser/src/browser.ts

```diff
@@ -76,6 +107,30 @@ export class FileBrowser extends SidePanel {
     this.crumbs = new BreadCrumbs({ model, translator });
     this.crumbs.addClass(CRUMBS_CLASS);
 
+    // The folder toolbar appears immediately below the breadcrumbs and above the directory listing.
+    const searcher = FilenameSearcher({
+      updateFilter: (
+        filterFn: (item: string) => Partial<IScore> | null,
+        query?: string
+      ) => {
+        this.model.setFilter(value => {
+          return filterFn(value.name.toLowerCase());
+        });
+      },
+      useFuzzyFilter: true,
+      placeholder: this._trans.__('Filter files by name'),
+      forceRefresh: true,
+      showIcon: false
+    });
+    searcher.addClass(FILTERBOX_CLASS);
+
+    this.folderToolbar = new Toolbar();
```

Comment:
Why is it named `folderToolbar`? Do we need it to be a toolbar, or can we just add this widget directly?

Reply:
As @jtpio suggests below, this could be made into a toolbar of its own, with a toolbar registry, although it isn't yet. Should it be?

Reply:
I guess it could be a toolbar if we want to add other buttons to configure the behaviour of the filter in the future. My first point was more about the "folder" part - it is not obvious why it is called "folder"

Reply:
I was trying to convey that it applies in the context of a folder (directory) and that it appears below the folder breadcrumbs in the UI. Open to other suggestions.

Reply:
While it does filter a single folder by default, with https://github.com/jupyterlab-contrib/jupyterlab-unfold it filters all (expanded) folders. It depends what should be in this toolbar. Since right now it only contains a filter, I would imagine `filterToolbar` to be a fitting name.

Thinking into the future I was imaging that it could also have buttons to toggle whether to filter in a single folder vs all folders, maybe whether to show hidden files, or whether to hide files in `.gitignore`. Maybe buttons are not a best solution but a dropdown with a set of options?